### ISSUE TITLE
Cross building from x64 to arm64 fails compiling using clang

### ISSUE THREAD

**Issue by AnInternetTroll** [dotnet/source-build#4566]
### Describe the Bug

When cross building from x64 to arm64 the process fails as clang "is not able to compile a simple test program."

### Steps to Reproduce

In the dotnet/dotnet repo on version v8.0.7 the following command should cause the issue 

`ROOTFS_DIR=/usr/aarch64-linux-gnu bash ./build.sh --clean-while-building --online --source-repository https://github.com/dotnet/dotnet --source-version 8be139ddde52d33e24c7d82f813248ff9fc54b97 -- -p:TargetArchitecture=arm64 -p:CrossBuild=true`

[The full void template is available on github (very much WIP)](https://github.com/AnInternetTroll/void-packages/blob/5fd90a4fb789076b35189838a76037f392b7ec21/srcpkgs/dotnet-sdk/template)

### Other Information

[The relevant output log of the build process can be found here](https://github.com/user-attachments/files/16664662/dotnet-sdk-aarch64-trimmed.txt)
(I have omitted the void-related stuff, such as installing dependencies, as well as the prep.sh output. If it's relevant I can paste it as well)

**Comment by omajid** [comment]
Some relevant bits from the log:

```
...
        The C compiler

          "/usr/bin/clang-17"

        is not able to compile a simple test program.

        It fails with the following output:
...
          /usr/bin/clang-17   -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=armv8-a  -g -I/usr/aarch64-linux-gnu/usr/include -ffile-prefix-map=/builddir/dotnet-sdk-8.0.7=.  -MD -MT CMakeFiles/cmTC_9081e.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_9081e.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_9081e.dir/testCCompiler.c.o -c /builddir/dotnet-sdk-8.0.7/src/runtime/artifacts/source-build/self/src/artifacts/obj/coreclr/linux.x64.Release/CMakeFiles/CMakeScratch/TryCompile-NWEw92/testCCompiler.c
          error: unknown target CPU 'armv8-a'
          note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
```

Could it be that the compiler being used can't target arm64 and the compiler that can't target arm64 isn't being picked up?

**Comment by AnInternetTroll** [comment]
Yeah, that seems to be the case. It looks as though void's clang can't cross compile, and they use cross-aarch64-linux-gnu instead as a compiler. If I install clang then dotnet's build script tries to use it and fails. If I don't install it then dotnet's build script will complain it can't find clang. In both cases the $CC variable is set to cross-aarch64-linux-gnu

I don't know how dependent dotnet is on clang, but would it be possible to use something else? Ideally though an env var or flag?

**Comment by omajid** [comment]
Can you try with `ROOTFS_DIR=/` ?

**Comment by omajid** [comment]
And also with `CROSSCOMPILE=1`

**Comment by omajid** [comment]
I haven't done any cross-builds for arm64 myself, but when I compare with ppc64le cross builds (which I do build):

I set some env vars:

- `CROSSCOMPILE=1`
- `ROOTFS_DIR=/`
- `PKG_CONFIG_PATH=/usr/lib/powerpc64le-linux-gnu/pkgconfig`

My build command is then literally:

`./build.sh --source-only --online --use-mono-runtime /p:TargetArchitecture=ppc64le /p:PortableBuild=true`

**Comment by MichaelSimons** [comment]
[Triage] This appears as a discussion with no actionable changes necessary within source-build.  Converting to a discussion.  Please open follow-up issues if any source changes are needed.
